host/xtest/regression_4000.c

```diff
@@ -2724,6 +2727,7 @@ struct xtest_ac_case {
 	XTEST_AC_CASE(level, algo, mode, vect, XTEST_AC_ECDSA_UNION(vect))
 
 static const struct xtest_ac_case xtest_ac_cases[] = {
+#if defined(CFG_CRYPTOLIB_NAME_mbedtls) &&  defined(MBEDTLS_RSA_NO_CRT)
```

Comment:
prefer:
`#if !defined(CFG_CRYPTOLIB_NAME_mbedtls) || defined(MBEDTLS_RSA_NO_CRT)`
so that these test case are still embedded when using libtomcryp as crypto lib.

Reply:
libtomcryp support RSA with CRT and RSA without CRT ，libtomcryp can handle both situations well，
but mbedtls Alternative support through macro MBEDTLS_RSA_NO_CRT control. xtest 4006 some case no provide CRT will run fail when enable mbedtls with CRT in optee-os.

> I'm not sure this is such a good idea. It's better to focus on the root of the problem in my opinion.

regression_4000.c:3904: ta_crypt_cmd_asymmetric_decrypt(c, &session, op, ((void *)0), 0, out_enc, out_enc_size, out, &out_size) has an unexpected value: 0xffff0006 = TEEC_ERROR_BAD_PARAMETERS, expected 0x0 = TEEC_SUCCESS
  regression_4006.1 FAILED
  regression_4006 FAILED

if enable mbedtls with CRT, it will check CRT as below code and return MBEDTLS_ERR_RSA_BAD_INPUT_DATA

in function 
rsa_check_context

```
 188 #if !defined(MBEDTLS_RSA_NO_CRT)
 189     /* Modular exponentiation for P and Q is only
 190      * used for private key operations and if CRT
 191      * is used. */
 192     if( is_priv &&
 193         ( mbedtls_mpi_cmp_int( &ctx->P, 0 ) <= 0 ||
 194           mbedtls_mpi_get_bit( &ctx->P, 0 ) == 0 ||
 195           mbedtls_mpi_cmp_int( &ctx->Q, 0 ) <= 0 ||
 196           mbedtls_mpi_get_bit( &ctx->Q, 0 ) == 0  ) )
 197     {
 198         return( MBEDTLS_ERR_RSA_BAD_INPUT_DATA );
 199     }
 200 #endif /* !MBEDTLS_RSA_NO_CRT */
```

Reply:
Yes, we've discussed that problem earlier. The problem is that OP-TEE cannot provide a consistent ABI with `MBEDTLS_RSA_NO_CRT` disabled. The job for xtest is to among other things highlight such problems.

By the way, how is the pull request to upstream mbedtls progressing?

Reply:
reply as below
“As I mentioned above, we need to schedule some resources to move this PR forward. This usually takes several months.“

detail 
https://github.com/ARMmbed/mbedtls/pull/4742

Reply:
Given that undefining `MBEDTLS_RSA_NO_CRT` is likely to break the GP tests it seems unlikely that we can accept such a change upstream.

Reply:
is there anything holding this PR back? 
Do I need to close this PR？
thanks.
@jenswi-linaro @etienne-lms @jforissier

Reply:
> is there anything holding this PR back?

I suppose you missed:
>> Given that undefining `MBEDTLS_RSA_NO_CRT` is likely to break the GP tests it seems unlikely that we can accept such a change upstream.

Reply:
I don't think this patch will break GP testing,it only reduce xtest-4006 without CRT test  when mbedtls with CRT enable in  optee-os. Other scenes were not affected.
I think that It is not possible to support this scenario by supporting dynamic checking of CRT parameters in the short term.
Because there's too much code involved.

Reply:
This repository is about testing upstream OP-TEE OS. If we undefine `MBEDTLS_RSA_NO_CRT` when compiling OP-TEE OS it will not pass the GP tests so we're not likely to merge such a patch, hence no need for this either.

Reply:
ok ，thanks.